src/main/java/pl/beata/zajecia22_zad1/UserController.java

```diff
@@ -0,0 +1,97 @@
+        this.userRepository = userRepository;
+    }
+
+    @ResponseBody
+    @RequestMapping("/addUserFromForm")
+    public String addUserFromForm(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko", defaultValue = "xxx") String surname, @RequestParam(name = "wiek", defaultValue = "-1") Integer age) {
+        if (name.equals("")) {
+            return "redirect:/err.html";
+        } else {
+            User newUser = new User(name, surname, age);
+            userRepository.add(newUser);
+            return "redirect:/success.html";
+        }
+    }
+
+
+    @RequestMapping("/add")
+    public String addUser(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko") String surname, @RequestParam(name = "wiek") Integer age) {
+        if (name.equals("")) {
+            return "err.html";
+        } else {
+            User newUser = new User(name, surname, age);
+            userRepository.add(newUser);
+            return "success.html";
+        }
+    }
+
+    @ResponseBody
+    @RequestMapping("/users")
+    public String getUsers() {
+        List<User> usersList = userRepository.getUsers();
+        String result = "";
+        for (User user : usersList) {
+            result += user.getName() + " " + user.getSurname() + " wiek: " + user.getAge() + "<br/>";
+        }
+        return result;
+    }
+
+    @ResponseBody
+    @RequestMapping("/filterUsers")
+    public String filterUsers(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko", defaultValue = "xxx", required = false) String surname, @RequestParam(name = "wiek", defaultValue = "-1", required = false) Integer age) {
+        List<User> usersList = getUsers(name, surname, age);
+
+        String result = "";
+        for (User user : usersList) {
+            result += user.getName() + " " + user.getSurname() + " wiek: " + user.getAge() + "<br/>";
+        }
+        return result;
+    }
+
+    @ResponseBody
+    @RequestMapping("/deleteUsers")
+    public String deleteUsers(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko", defaultValue = "xxx", required = false) String surname, @RequestParam(name = "wiek", defaultValue = "-1", required = false) Integer age) {
+        List<User> usersToBeRemoved = getUsers(name, surname, age);
+        userRepository.getUsers().removeAll(usersToBeRemoved);
+
+        String result = "";
+        for (User user : userRepository.getUsers()) {
+            result += user.getName() + " " + user.getSurname() + " wiek: " + user.getAge() + "<br/>";
+        }
+        return result;
+    }
+
+    private List<User> getUsers(@RequestParam(name = "imie") String name, @RequestParam(name = "nazwisko", defaultValue = "xxx", required = false) String surname, @RequestParam(name = "wiek", defaultValue = "-1", required = false) Integer age) {
+        List<User> usersList = userRepository.getUsers();
+
+        if (!name.equals("")) {
```

Comment:
Alternatywnie - można to było zrobić sprawdzając na każdym użytkowniku w repozytorium czy dany atrybut został wprowadzony, oraz czy pasuje do tego atrybutu który użytkownik podał. Następnie można takie coś sprawdzić dla 3 atrybutów i podjąć decyzję o tym czy użytkownik z repozytorium spełnia wymagania czy też nie. Kod mógłby wyglądać następująco:
```
private boolean checkSearchConditionsForUser(String imie, String nazwisko, Integer wiek, User user) {
        boolean nameOk = true;
        boolean surnameOk = true;
        boolean ageOk = true;

        if(!StringUtils.isEmpty(imie) && !imie.equals(user.getName())) {
            nameOk = false;
        }

        if(!StringUtils.isEmpty(nazwisko) && !nazwisko.equals(user.getLastName())) {
            surnameOk = false;
        }

        if(wiek != null && wiek != user.getAge()) {
            ageOk = false;
        }

        return nameOk && surnameOk && ageOk;
    }
```